src/main/java/org/spongepowered/common/mixin/core/item/inventory/TraitInventoryAdapter.java

```diff
@@ -132,4 +136,33 @@ protected final EmptyInventory emptyInventory() {
         this.getInventory().clear();
     }
 
+    @Override
+    public PluginContainer getPlugin() {
+        Object base = this;
+        if (this instanceof CarriedInventory) {
+            Optional<?> carrier = ((CarriedInventory<?>) this).getCarrier();
+            if (carrier.isPresent()) {
+                base = carrier.get();
+            }
+        }
+        if (base instanceof TileEntity) {
+            String id = ((TileEntity) this).getBlock().getType().getId();
+            String pluginId = id.substring(0, id.indexOf(":"));
+            return Sponge.getPluginManager().getPlugin(pluginId)
+                    .orElseThrow(() -> new AssertionError("Missing plugin " + pluginId + " for block " + id));
+        }
+        if (base instanceof Entity) {
+            String id = ((Entity) this).getType().getId();
+            String pluginId = id.substring(0, id.indexOf(":"));
+            return Sponge.getPluginManager().getPlugin(pluginId)
+                    .orElseThrow(() -> new AssertionError("Missing plugin " + pluginId + " for entity " + id + " (" + getClass().getName() + ")"));
+        }
+        if (base instanceof SpongeUser) {
+            return SpongeImpl.getMinecraftPlugin();
```

Comment:
I decided to use the Minecraft plugin for consistency with the `Player` inventory; they're conceptually the same, so IMO they should have the same owner. I'm open to changing it, however.